py/desispec/exposure_qa.py

```diff
@@ -0,0 +1,377 @@
+                worst_rdnoise = max(worst_rdnoise,rdnoise)
+                worst_rdnoise_per_petal = max(worst_rdnoise_per_petal,rdnoise)
+                if rdnoise > max_rdnoise :
+                    log.warning("readnoise is bad in camera {} amplifier {} : {}".format(camera,amp,rdnoise))
+                    readnoise_is_bad = True
+            petalqa_table["WORSTREADNOISE"][petal]=worst_rdnoise_per_petal
+
+            if readnoise_is_bad :
+
+                rdnoise_left  = max(head['OBSRDNA'],head['OBSRDNC'])
+                rdnoise_right = max(head['OBSRDNB'],head['OBSRDND'])
+
+                log.warning("readnoise is bad in at least one amplifier, flag affected fibers")
+                psf_filename=findfile('psf',night,expid,camera,specprod_dir=specprod_dir)
+                tset = read_xytraceset(psf_filename)
+                twave=np.linspace(tset.wavemin,tset.wavemax,20)
+                xtrans=float(head['CCDSIZE'].split(',')[0])/2.
+                xfiber=tset.x_vs_wave(fiber=np.arange(tset.nspec),wavelength=twave)[:,0]
+                if rdnoise_left>max_rdnoise :
+                    fiberqa_table['QAFIBERSTATUS'][entries[xfiber<xtrans]] |= bad_rdnoise_mask
+                elif rdnoise_right>max_rdnoise :
+                    fiberqa_table['QAFIBERSTATUS'][entries[xfiber>=xtrans]] |= bad_rdnoise_mask
+
+        # masks
+        ################
+        bad_positions_mask = fibermask.mask(qa_params["bad_positions_mask"]) # only positioning issues
+        bad_fibers_mask    = fibermask.mask(qa_params["bad_qafstatus_mask"]) # all possible issues
+
+
+        # checking statistics of positioning
+        ####################################################################
+        n_bad_positions = np.sum((fiberqa_table['QAFIBERSTATUS'][entries]&bad_positions_mask)>0)
+        if n_bad_positions > qa_params["max_frac_of_bad_positions_per_petal"]*entries.size :
+            log.warning("petal #{} has {} fibers with bad positions".format(petal,n_bad_positions))
+            fiberqa_table['QAFIBERSTATUS'][entries] |= fibermask.mask("BADPETALPOS")
+
+        petalqa_table["NGOODPOS"][petal]=np.sum((fiberqa_table['QAFIBERSTATUS'][entries]&bad_positions_mask)==0)
+
+
+        # checking standard stars
+        ####################################################################
+        stdstars_filename = findfile("stdstars",night,expid,spectrograph=spectro,specprod_dir=specprod_dir)
+        if os.path.isfile(stdstars_filename) :
+            t = fitsio.read(stdstars_filename,'METADATA')
+            # SNR cut is same as in stdstars.py, this is redundant,
+            # but for clarity on the selection, I repeat the cuts here
+            # CHI2DOF and color cut are used in flux calibration
+            # generous color cut here.
+            good=(t["CHI2DOF"]<2.)&(t["BLUE_SNR"]>=4.)
+            if "MODEL_G-R" in t.dtype.names :
+                good &= (np.abs(t["MODEL_G-R"]-t["DATA_G-R"])<0.3)
```

Comment:
Our favorite gotcha: off-footprint Gaia-std only tiles like TILEID 80715 observed 20210115/00072498, have stdstars*.fits METADATA columns DATA_GAIA-BP-RP and MODEL_GAIA-BP-RP instead of DATA_G-R and MODEL_G-R, thus this QA will crash.  See scripts/stdstars.py around line 594.

At minimum we should catch the Gaia special case here, but better if this could be abstracted into a way that we won't keep tripping on it.

Reply:
The code tests for "MODEL_G-R" and will not do this color test for GAIA magnitudes (which are  outside of the main footprint and sometimes affected by large extinction corrections).

Reply:
Got it; not applying cut on Gaia is purposeful, and I was wrong about this crashing.  Looks fine, thanks for confirming.